src/queues.c

```diff
@@ -133,6 +133,9 @@ static bool queues_notification_is_finished(struct notification *n, struct dunst
 {
         assert(n);
 
+        if (n->skip_display && !n->redisplayed)
+                return true;
```

Comment:
This hunk seems redundant, the notifications never see the displayed queue but skip directly from waiting to history. Was there a reason for adding this?

Reply:
It may seem redundant, but actually it isn't. For the case, that an already displayed notification gets replaced, this would trigger the immediate removal.